lib/deep-mark-required/index.ts

```diff
@@ -0,0 +1,30 @@
+import { AreNonDistributiveEqual } from "../are-non-distributive-equal";
+import { HasParsablePath } from "../has-parsable-path";
+import { MarkRequired } from "../mark-required";
+import { Paths } from "../paths";
+import { Prettify } from "../prettify";
+import { UnionToTuple } from "../union-to-tuple";
+
+type RecursiveDeepMarkRequiredSinglePath<Type, KeyPath> = KeyPath extends `${infer PropertyKey}.${infer RestKeyPath}`
+  ? {
+      [Key in keyof Type]: AreNonDistributiveEqual<Key, PropertyKey> extends true
```

Comment:
```suggestion
      [Key in keyof Type]: Key extends PropertyKey
```

<br />

Is there really a need for `AreNonDistributiveEqual` because neither `Key` nor `PropertyKey` would be a union here.

Reply:
Agree, addressed it in https://github.com/ts-essentials/ts-essentials/pull/408/commits/ca6055ce581992beb5d07a94011426aeec437ee7